### PR TITLE
tauri: hardening: more validation for commands

### PR DESCRIPTION
**Pull request by WofWca**
See individual commits for descriptions.

### CODE REVIEW

File: packages/target-tauri/src-tauri/src/temp_file.rs [hunk 1]
```diff
@@ -163,23 +176,41 @@ pub(crate) async fn remove_temp_file(app: AppHandle, path: SafePathBuf) -> Resul
...
+
     {
-        return Err(Error::SourcePathOutsideOfBlobsDir);
+        let maybe_blobs_dir = source_path
+            .parent()
+            .ok_or(Error::SourcePathOutsideOfBlobsDir)?;
+        if maybe_blobs_dir
+            .file_name()
+            .ok_or(Error::SourcePathOutsideOfBlobsDir)?
+            != "dc.db-blobs"
```
Comment (Simon-Laux) [inline packages/target-tauri/src-tauri/src/temp_file.rs:197]: I choose the ends with intentional as I thought someone might possible have an db with a different name, though thinking about it that edge-case will probably not happen, so checking for exact name is also fine

**Review by Simon-Laux — APPROVED** [review]
clippy has 2 issues, but besides that looks good, didn't test
